operate/client/src/modules/utils/flowNodeSelection.test.ts

```diff
@@ -114,3 +117,130 @@ describe('getSelectedRunningInstanceCount', () => {
     expect(result).toBe('Start Event');
   });
 });
+
+describe('selectAdHocSubProcessInnerInstance', () => {
+  beforeEach(() => {
+    flowNodeSelectionStore.reset();
+    flowNodeInstanceStore.reset();
+  });
+
+  const createMockFlowNodeInstance = (overrides = {}) => ({
+    id: 'mock-id',
+    type: 'SERVICE_TASK' as const,
+    state: 'ACTIVE' as const,
+    flowNodeId: 'mock-flow-node',
+    startDate: '2020-08-18T12:07:33.953+0000',
+    endDate: null,
+    treePath: 'mock/path',
+    sortValues: ['1606300828415', 'mock-id'] as [string, string],
+    ...overrides,
+  });
+
+  it('should select the first child when children are already loaded', () => {
+    const mockRootNode = {flowNodeId: 'root', flowNodeInstanceId: 'root-1'};
+    const mockInnerInstance = createMockFlowNodeInstance({
+      id: 'inner-1',
+      flowNodeId: 'adhoc-subprocess',
+      type: 'AD_HOC_SUB_PROCESS_INNER_INSTANCE',
+    });
+
+    const mockChild = createMockFlowNodeInstance({
+      id: 'child-1',
+      flowNodeId: 'task-1',
+      type: 'SERVICE_TASK',
+    });
+
+    // Mock that children are already loaded
+    const getVisibleChildNodesSpy = vi
+      .spyOn(flowNodeInstanceStore, 'getVisibleChildNodes')
+      .mockReturnValue([mockChild]);
```

Comment:
We don't mock unless it's absolutely necessary. Please rewrite the tests so they work without mocking